docs/read-write.ipynb

```diff
@@ -0,0 +1,312 @@
+      "Key(offsets={'lat': 0, 'lon': 0, 'time': 1000}, vars=None)\n",
+      "  with <xarray.Dataset data_vars=['air'] dims={'lat': 25, 'time': 1000, 'lon': 53}>\n",
+      "Key(offsets={'lat': 0, 'lon': 0, 'time': 2000}, vars=None)\n",
+      "  with <xarray.Dataset data_vars=['air'] dims={'lat': 25, 'time': 920, 'lon': 53}>\n"
+     ]
+    }
+   ],
+   "source": [
+    "with beam.Pipeline() as p:\n",
+    "    p | xbeam.DatasetToChunks(ds, chunks={'time': 1000}) | beam.MapTuple(print_summary)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2f0e5efb",
+   "metadata": {},
+   "source": [
+    "Importantly, xarray datasets fed into `DatasetToChunks` **can be lazy**, with data not already loaded eagerly into NumPy arrays. When you feed lazy datasets into `DatasetToChunks`, each individual chunk will be indexed and evaluated separately on Beam workers.\n",
+    "\n",
+    "This pattern allows for leveraging Xarray's builtin dataset loaders (e.g., `open_dataset()` and `open_zarr()`) for feeding arbitrarily large datasets into Xarray-Beam.\n",
+    "\n",
+    "For best performance, set `chunks=None` when opening datasets and then _explicitly_ provide chunks in `DatasetToChunks`:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 47,
+   "id": "8a0d0091",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "WARNING:root:Make sure that locally built Python SDK docker image has Python 3.9 interpreter.\n"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Key(offsets={'lat': 0, 'lon': 0, 'time': 0}, vars=None)\n",
+      "  with <xarray.Dataset data_vars=['air'] dims={'time': 1000, 'lat': 25, 'lon': 53}>\n",
+      "Key(offsets={'lat': 0, 'lon': 0, 'time': 1000}, vars=None)\n",
+      "  with <xarray.Dataset data_vars=['air'] dims={'time': 1000, 'lat': 25, 'lon': 53}>\n",
+      "Key(offsets={'lat': 0, 'lon': 0, 'time': 2000}, vars=None)\n",
+      "  with <xarray.Dataset data_vars=['air'] dims={'time': 920, 'lat': 25, 'lon': 53}>\n"
+     ]
+    }
+   ],
+   "source": [
+    "# write data into the distributed Zarr format\n",
+    "ds.chunk({'time': 1000}).to_zarr('example-data.zarr', mode='w')\n",
+    "\n",
+    "# load it with zarr\n",
+    "on_disk = xarray.open_zarr('example-data.zarr', chunks=None)\n",
+    "\n",
+    "with beam.Pipeline() as p:\n",
+    "    p | xbeam.DatasetToChunks(on_disk, chunks={'time': 1000}) | beam.MapTuple(print_summary)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "de622acb",
+   "metadata": {},
+   "source": [
+    "`chunks=None` tells Xarray to use its builtin lazy indexing machinery, instead of using Dask. This is advantageous because datasets using Xarray's lazy indexing are serialized much more compactly (via [pickle](https://docs.python.org/3/library/pickle.html)) when passed into Beam transforms."
```

Comment:
I really appreciate this paragraph, as someone who is more familiar with Beam than the XArray ecosystem. A key question I'd want to find out about reading the docs is if chunks were serializable.

Reply:
>  A key question I'd want to find out about reading the docs is if chunks were serializable.

did I answer this question here or are you still wondering? :)

Reply:
Yes, you answered the question here.